src/library/socket.cpp

```diff
@@ -120,6 +115,11 @@ int UdpSocket::Connect(const std::string &aHost, uint16_t aPort)
 {
     auto portStr = std::to_string(aPort);
 
+    if (mNetCtx.fd >= 0)
+    {
+        mbedtls_net_free(&mNetCtx);
+    }
```

Comment:
nit: If `mbedtls_net_free` is idempotent, maybe we can just:
```suggestion
    mbedtls_net_free(&mNetCtx);
```